DataFormats/L1TCalorimeterPhase2/interface/CaloCrystalCluster.h

```diff
@@ -0,0 +1,152 @@
+#ifndef DataFormats_L1TCalorimeterPhase2_CaloCrystalsCluster_h
+#define DataFormats_L1TCalorimeterPhase2_CaloCrystalsCluster_h
+
+#include <vector>
+#include <map>
+#include <string>
+#include <algorithm>
+#include "DataFormats/L1Trigger/interface/L1Candidate.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+namespace l1tp2 {
+
+  class CaloCrystalCluster : public l1t::L1Candidate {
+  public:
+    CaloCrystalCluster()
+        : l1t::L1Candidate(),
+          calibratedPt_(0.),
+          hovere_(0.),
+          iso_(0.),
+          PUcorrPt_(0.),
+          bremStrength_(0.),
+          e2x2_(0.),
+          e2x5_(0.),
+          e3x5_(0.),
+          e5x5_(0.),
+          standaloneWP_(0.),
+          electronWP98_(0.),
+          photonWP80_(0.),
+          electronWP90_(0.),
+          looseL1TkMatchWP_(0.),
+          stage2effMatch_(0.){};
+
+    CaloCrystalCluster(const PolarLorentzVector &p4,
+                       float calibratedPt,
+                       float hovere,
+                       float iso,
+                       DetId seedCrystal,
+                       float PUcorrPt = 0.,
+                       float bremStrength = 0.,
+                       float e2x2 = 0.,
+                       float e2x5 = 0.,
+                       float e3x5 = 0.,
+                       float e5x5 = 0.,
+                       bool standaloneWP = false,
+                       bool electronWP98 = false,
+                       bool photonWP80 = false,
+                       bool electronWP90 = false,
+                       bool looseL1TkMatchWP = false,
+                       bool stage2effMatch = false)
+        : l1t::L1Candidate(p4),
+          calibratedPt_(calibratedPt),
+          hovere_(hovere),
+          iso_(iso),
+          seedCrystal_(seedCrystal),
+          PUcorrPt_(PUcorrPt),
+          bremStrength_(bremStrength),
+          e2x2_(e2x2),
+          e2x5_(e2x5),
+          e3x5_(e3x5),
+          e5x5_(e5x5),
+          standaloneWP_(standaloneWP),
+          electronWP98_(electronWP98),
+          photonWP80_(photonWP80),
+          electronWP90_(electronWP90),
+          looseL1TkMatchWP_(looseL1TkMatchWP),
+          stage2effMatch_(stage2effMatch){};
+
+    virtual ~CaloCrystalCluster(){};
+    inline float calibratedPt() const { return calibratedPt_; };
+    inline float hovere() const { return hovere_; };
+    inline float isolation() const { return iso_; };
+    inline float PUcorrPt() const { return PUcorrPt_; };
+    inline float bremStrength() const { return bremStrength_; };
+    inline DetId seedCrystal() const { return seedCrystal_; };
+    void SetCrystalPtInfo(std::vector<float> info) {
+      std::sort(info.begin(), info.end());
+      std::reverse(info.begin(), info.end());
+      crystalPt_ = std::move(info);
+    };
+    void SetExperimentalParams(const std::map<std::string, float> &params) { experimentalParams_ = params; };
```

Comment:
2.8 in http://cms-sw.github.io/cms_coding_rules.html
`Start method names with lowercase, use upper case initials for following words, e.g. collisionPoint().`